src/Tools/dotnet-monitor/EndpointInfo/ServerEndpointInfoSource.cs

```diff
@@ -82,6 +84,23 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
             if (_portOptions.ConnectionMode == DiagnosticPortConnectionMode.Listen)
             {
+                if (_portOptions.GetClearEndpointOnStartup() &&
+                   !RuntimeInformation.IsOSPlatform(OSPlatform.Windows) &&
+                    File.Exists(_portOptions.EndpointName))
+                {
+                    // In some circumstances stale files from previous instances of dotnet-monitor cause
+                    // the new instance to fail binding. We need to delete the file in this situation.
+                    try
+                    {
+                        _logger.EndpointDeleteAttempt(_portOptions.EndpointName);
+                        File.Delete(_portOptions.EndpointName);
+                    }
```

Comment:
Consider: We always say that we're attempting the delete, but only say if we failed (and say nothing if we succeed) - do we want to log a message (lower than Warning level) that informs people that the deletion succeeded? May be unnecessary, but was just a thought.

Reply:
This would be reasonable, but I mostly added these messages for our purposes since this is reactionary to a specific customer issue. I don't think any additional lines would be necessary for us to understand the logs around this issue.